jax/random.py

```diff
@@ -215,17 +218,43 @@ def threefry_2x32(keypair, count):
     msg = "threefry_2x32 requires uint32 arguments, got {}"
     raise TypeError(msg.format([lax.dtype(x) for x in [key1, key2, count]]))
 
-  odd_size = count.size % 2
-  if odd_size:
-    x = list(np.split(np.concatenate([count.ravel(), onp.uint32([0])]), 2))
-  else:
-    x = list(np.split(count.ravel(), 2))
+  assert count.ndim == 2
+  count_high, count_low = np.transpose(count)
 
-  x = threefry2x32_p.bind(key1, key2, x[0], x[1])
-  out = np.concatenate(x)
+  out = np.stack(threefry2x32_p.bind(key1, key2, count_high, count_low), -1)
   assert out.dtype == onp.uint32
-  return lax.reshape(out[:-1] if odd_size else out, count.shape)
 
+  return out
+
+def _interleave(arr_1, arr_2):
```

Comment:
Could this interleaving, which replaces what was previously concatenation, cause a performance regression?